src/sysdir.c

```diff
@@ -34,12 +37,63 @@ static int git_sysdir_guess_system_dirs(git_buf *out)
 #endif
 }
 
+#ifndef GIT_WIN32
+static int get_passwd_home(git_buf *out, uid_t uid)
+{
+	struct passwd pwd, *pwdptr;
+	char *buf = NULL;
+	long buflen;
+	int error;
+
+	assert(out);
+
+	if ((buflen = sysconf(_SC_GETPW_R_SIZE_MAX)) == -1)
+		buflen = 1024;
+
+	do {
+		buf = git__realloc(buf, buflen);
+		buflen *= 2;
+		error = getpwuid_r(uid, &pwd, buf, buflen, &pwdptr);
+	} while (error == ERANGE);
```

Comment:
Let's put a max on this, just in case?  😁

Reply:
Sure, why not. Restricted it to 8kb, which I'd hope should be plenty.